client/src/client_error.rs

```diff
@@ -26,6 +26,8 @@ pub enum ClientErrorKind {
     TransactionError(#[from] TransactionError),
     #[error(transparent)]
     FaucetError(#[from] FaucetError),
+    #[error(transparent)]
+    ConnectionError(#[from] ConnectionError),
```

Comment:
this variant is specific to `quinn` and its name needs to communicate that.

Reply:
i'd be tempted to create a new enum for quinn errors some where and map all their ZST-based errors to that, use it here as `QuinnError(#[from] QuinnError)`